main/ejml-dsparse/test/org/ejml/sparse/csc/mult/TestMatrixMatrixMultWithSemiRing_DSCC.java

```diff
@@ -0,0 +1,141 @@
+    public void mult_AT_B(String desc, DMatrixSparseCSC matrix, DMatrixSparseCSC otherMatrix) {
+        DSemiRing semiRing = DSemiRings.PLUS_TIMES;
+
+        DMatrixSparseCSC found = CommonOpsWithSemiRing_DSCC.multTransA(matrix, otherMatrix, null, semiRing, null, null);
+        DMatrixSparseCSC expected = CommonOps_DSCC.multTransA(matrix, otherMatrix, null, null, null);
+
+        EjmlUnitTests.assertEquals(expected, found);
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("sparseMatrixSources")
+    public void mult_A_BT(String desc, DMatrixSparseCSC matrix, DMatrixSparseCSC otherMatrix) {
+        DSemiRing semiRing = DSemiRings.PLUS_TIMES;
+
+        DMatrixSparseCSC found = CommonOpsWithSemiRing_DSCC.multTransB(matrix, otherMatrix, null, semiRing, null, null);
+        DMatrixSparseCSC expected = CommonOps_DSCC.multTransB(matrix, otherMatrix, null, null, null);
+
+        EjmlUnitTests.assertEquals(expected, found);
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("sparseMatrixSources")
+    public void elementMult(String desc, DMatrixSparseCSC matrix, DMatrixSparseCSC otherMatrix) {
+        DSemiRing semiRing = DSemiRings.PLUS_TIMES;
+
+        DMatrixSparseCSC found = CommonOpsWithSemiRing_DSCC.elementMult(matrix, otherMatrix, null, semiRing, null, null);
+        DMatrixSparseCSC expected = CommonOps_DSCC.elementMult(matrix, otherMatrix, null, null, null);
+
+        EjmlUnitTests.assertEquals(expected, found);
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("sparseMatrixSources")
+    public void add(String desc, DMatrixSparseCSC matrix, DMatrixSparseCSC otherMatrix) {
+        DSemiRing semiRing = DSemiRings.PLUS_TIMES;
+
+        DMatrixSparseCSC found = CommonOpsWithSemiRing_DSCC.add(1, matrix, 1, otherMatrix, null, semiRing, null, null);
+        DMatrixSparseCSC expected = CommonOps_DSCC.add(1, matrix, 1, otherMatrix, null, null, null);
+
+        EjmlUnitTests.assertEquals(expected, found);
+    }
+
+    private static Stream<Arguments> sparseVectorMatrixMultSources() {
+        return Stream.of(
+                // expected entries for (0, 0) and (0, 2)
+                Arguments.of("PLUS, TIMES", DSemiRings.PLUS_TIMES, new double[]{0.1, 0.5}),
+                Arguments.of("OR, AND", DSemiRings.OR_AND, new double[]{1, 1}),
+                Arguments.of("MIN, PLUS", DSemiRings.MIN_PLUS, new double[]{0.7, 0.9}),
+                Arguments.of("MAX, PLUS", DSemiRings.MAX_PLUS, new double[]{0.7, 1.1}),
+                Arguments.of("MIN, TIMES", DSemiRings.MIN_TIMES, new double[]{0.1, 0.2}),
+                Arguments.of("MAX, MIN", DSemiRings.MAX_MIN, new double[]{0.2, 0.5})
+        );
+    }
+
+    private static Stream<Arguments> sparseMatrixSources() {
+        Random rand = new Random(42);
+        DMatrixSparseCSC sparseMatrix = RandomMatrices_DSCC.rectangle(10, 10, 15, rand);
+        DMatrixSparseCSC denseMatrix = RandomMatrices_DSCC.rectangle(10, 10, 90, rand);
```

Comment:
By passing in the same matrix in `sparse sparse` and `dense dense` cases could you be testing a very trivial case accidentally? I do think that's a good test case but it also might be worth having another different sparse and dense matrix. Also calling a CSC matrix dense threw me for a second.

Reply:
I will add another sparse and dense matrix.
I hope calling it `denseCSC`, will minimize the confusion.